tools/wpt/virtualenv.py

```diff
@@ -100,9 +103,33 @@ def activate(self):
             # https://github.com/web-platform-tests/wpt/issues/27377
             # https://github.com/python/cpython/pull/9516
             os.environ.pop('__PYVENV_LAUNCHER__', None)
-        path = os.path.join(self.bin_path, "activate_this.py")
-        with open(path) as f:
-            exec(f.read(), {"__file__": path})
+
+        # Setup the path and site packages as if we'd launched with the virtualenv active
+        bin_dir = os.path.join(self.path, "bin")
+        os.environ["PATH"] = os.pathsep.join([bin_dir] + os.environ.get("PATH", "").split(os.pathsep))
+        os.environ["VIRTUAL_ENV"] = self.path
+
+        prev_length = len(sys.path)
+
+        schemes = set(sysconfig.get_scheme_names())
+        if "venv" in schemes:
+            scheme = "venv"
+        else:
+            scheme = "nt_user" if os.name == "nt" else "posix_user"
+        sys_paths = sysconfig.get_paths(scheme)
+        data_path = sys_paths["data"]
+        added = set()
+        for key in ["purelib", "platlib"]:
+            host_path = Path(sys_paths[key])
+            relative_path = host_path.relative_to(data_path)
+            site_dir = os.path.normpath(os.path.normcase(Path(self.path) / relative_path))
```

Comment:
Other reviewers: from https://docs.python.org/3/library/pathlib.html#basic-use it seems like the division operator can be used to join paths.